src/blinker.py

```diff
@@ -1,25 +1,33 @@
+import uasyncio
 from machine import Pin
 
 from sensor import Sensor
 
 
 class Blinker(Sensor):
 
-    def __init__(self, name, pin):
+    def __init__(self, name, pin, blink_duration=5):
         super().__init__(name)
 
         # we will need the LED pin later for some blinking.
         self.led = Pin(pin, Pin.OUT)
+        self.blink_duration = blink_duration
 
-    def measure(self):
-        if self.led.value() == 0:
-            self.led.value(1)
-        else:
-            self.led.value(0)
+    async def measure(self):
+        # turns on the led
+        self.led.value(0)
+        loop = uasyncio.get_event_loop()
+        loop.create_task(self._reset_blink())
```

Comment:
This should run in the background now. Create a task on the event loop but return here immediately